File: php/date_math_add.php
<?php
if ($argc != 3) {
    echo "Usage: php " . $argv[0] . " <seconds_to_add> <iterations>\n";
    exit(1);
}

$seconds_to_add = (int)$argv[1];
$iterations = (int)$argv[2];

$start = microtime(true);

for ($i = 0; $i < $iterations; $i++) {
    $time = time();
    $time += $seconds_to_add;
}

$end = microtime(true);
echo sprintf('%.12f', $end - $start) . "\n";

File: c/date_math_add.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/time.h>
#include <time.h>

int main(int argc, char *argv[]) {
    if (argc != 3) {
        fprintf(stderr, "Usage: %s <seconds_to_add> <iterations>\n", argv[0]);
        return 1;
    }

    int seconds_to_add = atoi(argv[1]);
    int iterations = atoi(argv[2]);

    struct timeval start, end;
    gettimeofday(&start, NULL);

    for (int i = 0; i < iterations; i++) {
        time_t current_time = time(NULL);
        current_time += seconds_to_add;
    }

    gettimeofday(&end, NULL);

    long seconds = (end.tv_sec - start.tv_sec);
    long micros = ((seconds * 1000000) + end.tv_usec) - (start.tv_usec);

    printf("%.12f\n", (double)micros / 1000000);

    return 0;
}
